### ISSUE TITLE
File Convert Two time !!!

### ISSUE THREAD

**Issue by Iranian-Atheist** [PHP-FFMpeg/PHP-FFMpeg#401]
Hi I Used this Code And I think each video is converted for two times if you refresh destination directory for many times (during converting) you can see file size increases to max. file size changes to 0 suddenlly and again increases to max.
it happens in task manager by cpu percentage too.
when i tested it with cmd and write command line i have not this problem !!!
sorry for bad english :))

```php
$ffmpeg = FFMpeg\FFMpeg::create(array(
        'ffmpeg.binaries'  => 'C:\ffmpeg\bin\ffmpeg.exe',
        'ffprobe.binaries' => 'C:\ffmpeg\bin\ffprobe.exe',
        'timeout'          => 3600,
        'ffmpeg.threads'   => 12,
    ), @$logger);
    $video = $ffmpeg->open($details['url']);
    $video
        ->filters()
        ->resize(new FFMpeg\Coordinate\Dimension($width, $height))
        ->framerate(new FFMpeg\Coordinate\Framerate(25), 50);
    $final = new FFMpeg\Format\Video\X264('aac', 'libx264');
    $final
        ->setKiloBitrate($bit_rate)
        ->setAudioChannels(2)
        ->setAudioKiloBitrate($audio_bit_rate);
    $video
        ->save($final, $details['dest_dir'].'/'.$details['name'].' ('.$height.'p).mp4');

**Comment by jens1o** [comment]
> and you can see this event in task manager by cpu percentage.

video editing in general is quite expensive

> when i test it with cmd and write command line i have not this problem !!!
can you share the command you use for that case?

>  refresh destination directory for many time (during converting) you can see file size increase to max suddenlly file size change to 0 and again increase to max.

That's how moving a file works. :/

**Comment by SGhazayi** [comment]
you can test this code with each video you want:

<?php
	$ffmpeg = FFMpeg\FFMpeg::create(array(
        'ffmpeg.binaries'  => 'C:\ffmpeg\bin\ffmpeg.exe',
        'ffprobe.binaries' => 'C:\ffmpeg\bin\ffprobe.exe',
        'timeout'          => 3600,
        'ffmpeg.threads'   => 12,
    ), @$logger);
    $video = $ffmpeg->open('input.mp4');
    $video
        ->filters()
        ->resize(new FFMpeg\Coordinate\Dimension(1920, 1080))
        ->framerate(new FFMpeg\Coordinate\Framerate(25), 50);
    $final = new FFMpeg\Format\Video\X264('aac', 'libx264');
    $final
        ->setKiloBitrate(2300)
        ->setAudioChannels(2)
        ->setAudioKiloBitrate(192);
    $video
        ->save($final, 'output.mp4');
?>

and you can test this command line. this command do same work that above code do:

ffmpeg -i input.mp4 -b:v 2300k -c:v libx264 -vf scale=1920:1080 -b:a 192k -c:a aac output.mp4

**Comment by jens1o** [comment]
Sorry, but personally I don't see the problem happen live.

**Comment by Miky94** [comment]
Hi SGhazayi, did you understand why this happens e how to solve this problem? Thanks

**Comment by SGhazayi** [comment]
Hi Miky94, Sorry i left this package and used exec() function.do you have this problem too?

**Comment by Miky94** [comment]
yes , but I noticed that with small video files it doesn't happen 

> refresh destination directory for many time (during converting) you can see file size increase to max suddenlly file size change to 0 and again increase to max.

> That's how moving a file works. :/

@jens1o 
I didn't understand this , is it normal during the converting process?

**Comment by Iranian-Atheist** [comment]
i think that happens too fast. at all this is very important bug each user's videos convert for two times!!! The server can be disabled

**Comment by jens1o** [comment]
@Miky94 Yes, it is. That's when ffmpeg is finished, closing the internal cursor(drop to 0) and then it's done.

**Comment by d4mn** [comment]
This issue still exists. Logger shows two conversions of same file.

`
[2020-01-08 11:03:15] local.INFO: ffprobe running command '/usr/bin/ffprobe' '-help' '-loglevel' 'quiet'  
[2020-01-08 11:03:16] local.INFO: ffprobe executed command successfully  
[2020-01-08 11:03:16] local.INFO: ffprobe running command '/usr/bin/ffprobe' '/home/dropbox/storage/app/public/videos/777/842/tHNKNOt2jKlZ3a1vxlhxDI3v1T46L6GMxR7vKPdu.qt' '-show_streams' '-print_format' 'json'  
[2020-01-08 11:03:16] local.INFO: ffprobe executed command successfully  
[2020-01-08 11:03:16] local.INFO: ffmpeg running command '/usr/bin/ffmpeg' '-y' '-i' '/home/dropbox/storage/app/public/videos/777/842/tHNKNOt2jKlZ3a1vxlhxDI3v1T46L6GMxR7vKPdu.qt' '-threads' '12' '-vcodec' 'libx264' '-acodec' 'aac' '-b:v' '1000k' '-refs' '6' '-coder' '1' '-sc_threshold' '40' '-flags' '+loop' '-me_range' '16' '-subq' '7' '-i_qfactor' '0.71' '-qcomp' '0.6' '-qdiff' '4' '-trellis' '1' '-b:a' '128k' '-pass' '1' '-passlogfile' '/dev/shm/ffmpeg-passes5e15b6f435285o4qcf/pass-5e15b6f4352e1' '/home/dropbox/storage/app/public/temp/converted.mp4'  
[2020-01-08 11:03:16] local.INFO: ffprobe running command '/usr/bin/ffprobe' '/home/dropbox/storage/app/public/videos/777/842/tHNKNOt2jKlZ3a1vxlhxDI3v1T46L6GMxR7vKPdu.qt' '-show_format' '-print_format' 'json'  
[2020-01-08 11:03:16] local.INFO: ffprobe executed command successfully  
[2020-01-08 11:03:22] local.INFO: ffmpeg executed command successfully  
[2020-01-08 11:03:22] local.INFO: ffmpeg running command '/usr/bin/ffmpeg' '-y' '-i' '/home/dropbox/storage/app/public/videos/777/842/tHNKNOt2jKlZ3a1vxlhxDI3v1T46L6GMxR7vKPdu.qt' '-threads' '12' '-vcodec' 'libx264' '-acodec' 'aac' '-b:v' '1000k' '-refs' '6' '-coder' '1' '-sc_threshold' '40' '-flags' '+loop' '-me_range' '16' '-subq' '7' '-i_qfactor' '0.71' '-qcomp' '0.6' '-qdiff' '4' '-trellis' '1' '-b:a' '128k' '-pass' '2' '-passlogfile' '/dev/shm/ffmpeg-passes5e15b6f435285o4qcf/pass-5e15b6f4352e1' '/home/dropbox/storage/app/public/temp/converted.mp4'  
[2020-01-08 11:03:35] local.INFO: ffmpeg executed command successfully  
`

**Comment by fogdingding** [comment]
- can use setPasses()
- src/FFMpeg/Format/Video/X264.php 

php-ffmpeg default pass is 2.
They said it would be better for the video conversion.
好像在某個地方有註解說，這樣的轉換後的影片會比較好。
